src/D2L.Bmx/Aws/AwsClient.cs

```diff
@@ -0,0 +1,75 @@
+using System.Text;
+using System.Web;
+using System.Xml;
+using Amazon.SecurityToken;
+using Amazon.SecurityToken.Model;
+
+namespace D2L.Bmx.Aws;
+
+public interface ICloudProvider<TRoleState> where TRoleState : IRoleState {
+	TRoleState GetRoles( string encodedSaml );
+	Task<Dictionary<string, string>> GetTokensAsync( TRoleState state, int selectedRoleIndex );
+}
+public class AwsClient : ICloudProvider<AwsRoleState> {
```

Comment:
* nit: blank line between top level blocks
* the class should also be internal

```suggestion
}

internal class AwsClient : IAwsClient {
```